MaxText/max_utils.py

```diff
@@ -398,12 +462,20 @@ def create_device_mesh(config, devices=None):
 
   if multi_slice_env:
     dcn_parallelism = fill_unspecified_mesh_axes(dcn_parallelism, num_slices, "DCN")
-    mesh = mesh_utils.create_hybrid_device_mesh(
+    if config.custom_mesh == "hybrid_ring_64x4":
+      assert config.ici_tensor_parallelism == 4, f"Invalid custom mesh chosen for mesh shape {ici_parallelism}"
```

Comment:
Can we make this more general - I think we might want this 64 x 4 mesh with expert parallelism = 4 (e.g. a lot of mixtral models only have 8 experts so we cant use the full 16 axis)

Reply:
asserting on fsdp now